src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs

```diff
@@ -126,6 +127,9 @@ public Test BuildFrom(IMethodInfo method, Test parentSuite)
                     needCombinatorial = false;
             }
 
+            // This check must be done before CombinatorialAttribute gets added to the builders collection
+            var shouldForceParameterization = builders.Count > 0;
```

Comment:
I could see valid future scenarios for builders without parameters. And for parameters without builders, we simply create the single test and let the usual error occur. I do not fully understand the comments written with the TODO so I'll leave this alone for now.